Sources/HTTP/HTTPHeadersAccessors.swift

```diff
@@ -0,0 +1,1229 @@
+    public mutating func set(acceptEncoding: Encoding, quality: Double?) {
+        self.storage[.acceptEncoding]?.removeAll()
+        self.add(acceptEncoding: acceptEncoding, quality: quality)
+    }
+    
+    /// Adds a new `Accept` header value
+    public mutating func add(acceptEncoding: Encoding, quality: Double?) {
+        if self.storage[.acceptEncoding] == nil {
+            self.storage[.acceptEncoding] = []
+        }
+        if let qualityDesc = quality.flatMap({ String(format: "%.1f", Double.minimum(0, Double.maximum($0, 1))) }) {
+            self.storage[.acceptEncoding]!.append("\(acceptEncoding.rawValue); q=\(qualityDesc)")
+        } else {
+            self.storage[.acceptEncoding]!.append(acceptEncoding.rawValue)
+        }
+    }
+    
+    // `Authorization` header value
+    public var authorization: HTTPHeaders.Authorization? {
+        get {
+            return self.storage[.authorization]?.first.flatMap(Authorization.init)
+        }
+        set {
+            self.storage[.authorization] = newValue.flatMap { [$0.description] }
+        }
+    }
+    
+    // `Cookie` header value
+    public var cookie: [HTTPCookie] {
+        let pairs: [(key: String, val: String)] = (self.storage[.cookie]?.first?.components(separatedBy: ";").flatMap { text in
+            let segments = text.components(separatedBy: "=")
+            guard let key = segments.first?.trimmingCharacters(in: .whitespaces), !key.isEmpty else {
+                return nil
+            }
+            let value = segments.dropFirst().joined(separator: "=")
+            return (key, value)
+            }) ?? []
+        
+        return pairs.flatMap {
+            // path should be set otherwise it will fail!
+            return HTTPCookie(properties: [.name : $0.key, .value: $0.val, .path: "/"])
+        }
+    }
+    
+    /// `If-Match` header etag value
+    public var ifMatch: [EntryTag] {
+        get {
+            return (self.storage[.ifMatch] ?? []).map(EntryTag.init)
+        }
+        set {
+            // TOCHECK: When there is a wildcard, other values should be ignored
+            if !newValue.isEmpty {
+                self.storage[.ifMatch] = newValue.map { $0.description }
+            } else {
+                self.storage[.ifMatch] = nil
+            }
+        }
+    }
+    
+    /// `If-None-Match` header etag value
+    public var ifNoneMatch: [EntryTag] {
+        get {
+            return (self.storage[.ifNoneMatch] ?? []).map(EntryTag.init)
+        }
+        set {
+            // TOCHECK: When there is a wildcard, other values should be ignored
+            if !newValue.isEmpty {
+                self.storage[.ifNoneMatch] = newValue.map { $0.description }
+            } else {
+                self.storage[.ifNoneMatch] = nil
+            }
+        }
+    }
+    
+    /// `If-Range` header etag value
+    public var ifRange: EntryTag? {
+        get {
+            return self.storage[.ifRange]?.first.flatMap(EntryTag.init)
```

Comment:
As of RFC7233 and RFC7232 `ifRange` could also be of the form of `Last-Modified` header. instantiating from EntryTag will not allow the later form.